File: app/src/main/java/com/waveneuro/ui/session/how_to/HowToActivity.java
package com.waveneuro.ui.session.how_to;

import android.graphics.Paint;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.viewpager.widget.ViewPager;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.tabs.TabLayout;
import com.waveneuro.R;
import com.waveneuro.ui.base.BaseActivity;
import com.waveneuro.ui.session.precautions.PrecautionsBottomSheet;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;


public class HowToActivity extends BaseActivity {
    @Inject
    HowToViewModel howToViewModel;

    @BindView(R.id.tv_skip_onboarding)
    TextView tvSkipOnboarding;

    @BindView(R.id.tv_page_indicator)
    TextView tvPageIndicator;

    @BindView(R.id.btn_got_it)
    MaterialButton btnGotIt;

    @BindView(R.id.tl_dots)
    TabLayout tlDots;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_how_to);

        activityComponent().inject(this);
        ButterKnife.bind(this);

        tvSkipOnboarding.setPaintFlags(tvSkipOnboarding.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
        ViewPager viewPager = (ViewPager) findViewById(R.id.viewpager);
        viewPager.setAdapter(new HowToAdapter(this));
        tlDots.setupWithViewPager(viewPager);


        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            public void onPageScrollStateChanged(int state) {}
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {}

            public void onPageSelected(int position) {
                tvPageIndicator.setText((position+1)+" of 6");
                if (position == 5) {
                    btnGotIt.setVisibility(View.VISIBLE);
                    tvSkipOnboarding.setText(R.string.list_of_precautions);
                    tvSkipOnboarding.setOnClickListener(v -> {
                        PrecautionsBottomSheet precautionsBottomSheet = PrecautionsBottomSheet.newInstance();
                        precautionsBottomSheet.show(getSupportFragmentManager(), "");
                    });

                } else {
                    btnGotIt.setVisibility(View.INVISIBLE);
                    tvSkipOnboarding.setText(R.string.skip_onboarding);
                    tvSkipOnboarding.setOnClickListener(v -> finish());

                }
            }
        });
    }

    @OnClick(R.id.btn_got_it)
    public void btnGotIt() {
        finish();
    }

    @OnClick(R.id.tv_skip_onboarding)
    public void onSkipClicked() {
        finish();
    }

}

File: app/src/main/java/com/waveneuro/ui/session/how_to/HowToActivity.kt
package com.waveneuro.ui.session.how_to

import android.content.Context
import android.content.Intent
import android.graphics.Paint
import android.os.Bundle
import android.view.View
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.waveneuro.R
import com.waveneuro.databinding.ActivityHowToBinding
import com.waveneuro.ui.base.activity.BaseActivity
import com.waveneuro.ui.session.precautions.PrecautionsBottomSheet

class HowToActivity : BaseActivity<ActivityHowToBinding>() {

    override fun initBinding(): ActivityHowToBinding = ActivityHowToBinding.inflate(layoutInflater)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setView()
    }

    private fun setView() {
        with(binding) {
            btnGotIt.setOnClickListener {
                finish()
            }
            tvSkipOnboarding.setOnClickListener {
                finish()
            }
            tvSkipOnboarding.paintFlags = tvSkipOnboarding.paintFlags or Paint.UNDERLINE_TEXT_FLAG
            val viewPager = findViewById<View>(R.id.viewpager) as ViewPager
            viewPager.adapter = HowToAdapter(this@HowToActivity)
            tlDots.setupWithViewPager(viewPager)
            viewPager.addOnPageChangeListener(object : OnPageChangeListener {
                override fun onPageScrollStateChanged(state: Int) {}
                override fun onPageScrolled(
                    position: Int,
                    positionOffset: Float,
                    positionOffsetPixels: Int
                ) {}
                override fun onPageSelected(position: Int) {
                    tvPageIndicator.text = (position + 1).toString() + " of 6"
                    if (position == 5) {
                        btnGotIt.visibility = View.VISIBLE
                        tvSkipOnboarding.setText(R.string.list_of_precautions)
                        tvSkipOnboarding.setOnClickListener {
                            val precautionsBottomSheet = PrecautionsBottomSheet.newInstance()
                            precautionsBottomSheet.show(supportFragmentManager, "")
                        }
                    } else {
                        btnGotIt.visibility = View.INVISIBLE
                        tvSkipOnboarding.setText(R.string.skip_onboarding)
                        tvSkipOnboarding.setOnClickListener { finish() }
                    }
                }
            })
        }
    }

    companion object {
        fun newIntent(context: Context) = Intent(context, HowToActivity::class.java)
    }

}